### PR TITLE
Image upload in Trip Create/Edit pages

### PR DESCRIPTION
**Pull request by samuraikun**
# Description
* This PR is uploading image in trip page

- ※ 🔔 Display time for image might be slow due to load directly images from Supabase Storage without CDN. 🙏 
  - If we use CDN, we need to upgrade Supabase Pro plan (not free)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Check List
- [x] Preview uploaded image
- [x] Upload image file to Supabase Storage
- [x] Update trips image_url column after uploaded file

### Create Trip
60360d3440ba52a5ff7383b203cfe44b

### Update Trip
dbb828e5e887c6d73225ca2c1a5d7efa

### CODE REVIEW

File: app/trip/components/trip-form.tsx [hunk 1]
```diff
@@ -147,8 +151,43 @@ export const TripForm = ({ tripDetails, tags, tripTags }: TripFormProps) => {
...
+                src={tripDetails?.image || imageSrc}
+                width="50%"
+                objectFit="cover"
+              />
+            )}
+            <Controller
+              name="uploaded_image_file"
+              control={control}
+              render={({ field: { onChange } }) => (
+                <PrimaryButton variant="outline" as="label">
```
Comment (kanatagu) [inline app/trip/components/trip-form.tsx]: Can you add cursor:pointer for hover? 🙏

File: app/trip/hooks/useTripUpdate.ts [hunk 2]
```diff
@@ -71,6 +73,10 @@ export const useTripUpdate = (
 
       await Promise.all([...deletePromises, ...createPromises])
 
+      if (data.uploaded_image_file && tripDetails) {
+        await uploadFile(data.uploaded_image_file, tripDetails.id)
+      }
+
```
Comment (kanatagu) [inline app/trip/hooks/useTripUpdate.ts]: After uploaded, it shows the form for a sec, and then redirect to details page. Could you fix it?

File: app/trip/components/trip-form.tsx [hunk 3]
```diff
@@ -147,8 +151,43 @@ export const TripForm = ({ tripDetails, tags, tripTags }: TripFormProps) => {
...
+                  Select Image
+                  <Input
+                    type="file"
+                    accept="image/*"
```
Comment (SachiGoto) [inline app/trip/components/trip-form.tsx:176]: Thank you so much for adding the image upload feature! it's looks really nice! I'm just wondering how you've managed image types and sizes.  I've noticed that it currently accepts various image formats here but I've also noticed that PDFs are not accepted, and there is a limit on the size of images or GIFs that can be inserted. I'm curious about where these settings are configured?

Comment (samuraikun) [reply app/trip/components/trip-form.tsx:176]: Thanks for your feedback!
I added two validations in zod schema below.
- Only image file
- File size is less than 10MB

https://github.com/tabi-memo/frontend/pull/32/commits/3d3237cae33ee00f1163411a9cd512d4da6e3490

File: app/trip/components/trip-form.tsx (hunk 1 above)
Comment (samuraikun) [reply app/trip/components/trip-form.tsx]: Thx! I did.
https://github.com/tabi-memo/frontend/pull/32/commits/3d3237cae33ee00f1163411a9cd512d4da6e3490#diff-16197c3b76acbf9dfbb1182d34e1a7aa8f25beff37b8bbd1080e041960183ff2R173

File: app/trip/hooks/useTripUpdate.ts (hunk 2 above)
Comment (samuraikun) [reply app/trip/hooks/useTripUpdate.ts]: @kanatagu  I didn't know the problem despite of researching 🤔
Could I skip it as another issue. 🙏

Comment (kanatagu) [reply app/trip/hooks/useTripUpdate.ts]: @samuraikun 
Yes No problem!

**Review by kanatagu — APPROVED** [review]

**Pull request merged**